rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.84"
```

Comment:
Right now CI always uses latest, so this is a deviation from that behaviour. I could also change this to `"stable"` if there's consensus about it.

Reply:
Did that here: https://github.com/logos-co/nomos-node/pull/1018/commits/3db2eb9eb0df9e70f858295018ce7bf239562976.